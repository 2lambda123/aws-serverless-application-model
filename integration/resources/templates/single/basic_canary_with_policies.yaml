Resources:
  # added the Artifact Bucket inside the template so that only the policies from the Policies property gets added to the default role
  ResultsBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
  CanaryWithPolicies:
    Type: AWS::Serverless::Canary
    Properties:
      Handler: pageLoadBlueprint.handler
      InlineCode: |
        const loadBlueprint = async function () {
            console.log("hello")
        };
        exports.handler = async () => {
            return await loadBlueprint();
        };
      Runtime: syn-nodejs-puppeteer-3.1
      ArtifactS3Location:
        Fn::Join:
          - ''
          - - s3://
            - Ref: ResultsBucket
      Schedule:
        Expression: rate(0 minutes)
      StartCanaryAfterCreation: true
      Policies:
        - CloudWatchPutMetricPolicy: { }