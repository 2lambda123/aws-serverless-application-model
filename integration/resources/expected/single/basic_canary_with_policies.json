[
  { "LogicalResourceId":"CanaryWithPolicies", "ResourceType":"AWS::Synthetics::Canary" },
  { "LogicalResourceId":"CanaryWithPoliciesRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"ResultsBucket", "ResourceType":"AWS::S3::Bucket" }
]